src/test/resources/regressions/features/adts/match-error1.gobra

```diff
@@ -0,0 +1,29 @@
+// Any copyright is dedicated to the Public Domain.
+// http://creativecommons.org/publicdomain/zero/1.0/
+
+package matcherror1
+
+// this test case checks that Gobra does not crash when
+// typing a match expression if the corresponding ADT type
+// is not well-defined.
+
+ghost type Trace adt {
+    RootEntry{}
+    SendEntry{
+        //:: ExpectedOutput(type_error)
+        msg InexistentType // unknown identifier
+        prev Trace
+    }
+}
+
+ghost
+decreases _ // assume termination instead of using `len(t)`, which would cause another type error
```

Comment:
the type error we would get here would be essentially the same we have below, right?